components/notification-sender-config/org.wso2.carbon.identity.notification.sender.tenant.config/src/main/java/org/wso2/carbon/identity/notification/sender/tenant/config/NotificationSenderManagementServiceImpl.java

```diff
@@ -551,4 +584,55 @@ private String getChannelTypeFromResource(Resource resource) {
                     .map(Attribute::getValue).orElse(DEFAULT_HANDLER_NAME);
         }
     }
+
+    /**
+     * Method to check whether the sender is allowed to delete.
+     *
+     * @param senderName    Name of the sender.
+     * @return  whether the sender is allowed to delete.
+     * @throws NotificationSenderManagementException If an error occurred while checking the sender.
+     */
+    private boolean checkAllowDelete(String senderName) throws NotificationSenderManagementException {
+
+        if (SENDERS.get(senderName) != null) {
+            String authenticatorId = new String(Base64.getUrlEncoder()
+                    .encode(SENDERS.get(senderName).getBytes(StandardCharsets.UTF_8)),
+                    StandardCharsets.UTF_8);
+            String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            try {
+                ConnectedAppsResult appsResult =
+                        NotificationSenderTenantConfigDataHolder.getInstance()
+                        .getApplicationManagementService()
+                        .getConnectedAppsForLocalAuthenticator(authenticatorId, tenantDomain, 1 , 0);
+                // If there are any connected apps, the sender cannot be deleted.
+                if (appsResult.getApps().size() > 0) {
+                    return false;
+                }
+                Resource resource = NotificationSenderTenantConfigDataHolder.getInstance().getConfigurationManager()
+                        .getResource(MY_ACCOUNT_SMS_RESOURCE_TYPE, MY_ACCOUNT_SMS_RESOURCE_NAME);
+                String smsOtpEnabled = resource.getAttributes().stream().filter(attribute -> attribute
+                        .getKey().equals("sms_otp_enabled")).collect(
+                        Collectors.toList()).get(0).getValue();
+                // If SMS OTP is enabled for my account, the sender cannot be deleted.
+                if (Boolean.parseBoolean(smsOtpEnabled)) {
+                    return false;
+                }
+            } catch (IdentityApplicationManagementException e) {
+                if (e instanceof IdentityApplicationManagementClientException && e.getErrorCode()
+                        .equals(Error.AUTHENTICATOR_NOT_FOUND.getCode())) {
+                    return true;
+                }
+                throw handleApplicationMgtException(e, ERROR_CODE_ERROR_DELETING_NOTIFICATION_SENDER,
+                        senderName);
+            } catch (ConfigurationManagementException e) {
+                if (e instanceof ConfigurationManagementClientException && e.getErrorCode()
+                        .equals(ERROR_CODE_RESOURCE_DOES_NOT_EXISTS.getCode())) {
+                    return true;
+                }
+                throw handleConfigurationMgtException(e, ERROR_CODE_ERROR_DELETING_NOTIFICATION_SENDER,
```

Comment:
From this method, we are not trying to delete the connection right? We are only evaluating whether we can delete the connection. So throwing an error like 'ERROR_CODE_ERROR_DELETING_NOTIFICATION_SENDER' is not acceptable. Please add the correct error code.

Reply:
Addressed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/194/commits/10270bd3aa6edac00812a637f0da28fe3792cd02